### PR TITLE
catalog: use separate sjb configs for different branches

### PR DESCRIPTION
**Pull request by jpeeler**
This also removes the skip_report line for the master branch as that one
seems to be stable now.

### CODE REVIEW

File: cluster/ci/config/prow/config.yaml [hunk 1]
```diff
@@ -355,15 +355,28 @@ presubmits:
...
-  - name: test_pull_request_origin_service_catalog
+  - name: test_pull_request_origin_service_catalog_39
     agent: jenkins
     labels:
       master: ci.openshift.redhat.com
     always_run: false
-    # TEMPORARY until this job is fixed to work
+    branches:
+    - release-3.9
+    # skip_report line will be deleted once testing is completed
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/config.yaml]: This is understood and doesn't warrant a comment

Comment (jpeeler) [reply cluster/ci/config/prow/config.yaml]: Mostly a note to myself, but I can remove it.

**Comment by stevekuznetsov** [comment]
Let me know when you'd like to merge this!

**Comment by jpeeler** [comment]
@stevekuznetsov now would be great!

**Pull request merged**